File: 0909/HyowonPark/비밀지도.js

// 1 0 / 0 1 / 1 1 은 1
// 0 0 만 0 

function solution(n, arr1, arr2) {
    result = [];
    for (i = 0; i < n; i++) {
        row = [];
        a = arr1[i].toString(2).padStart(n, '0').split('');
        b = arr2[i].toString(2).padStart(n, '0').split('');
        
        for (j = 0; j < n; j++) {
            if (a[j] == '0'&& b[j] == '0') {
                row.push(" ");
            } else {
                row.push("#")
            }
        }

        result.push(row.join(''));
    }
    
    
    return result;
}

File: 0909/AyoungKim/비밀지도.java
import java.util.*;

class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];

        for (int i = 0; i < n; i++) {
            StringBuilder sb = new StringBuilder();
            String spot1 = String.format("%" + n + "s", String.valueOf(Integer.toBinaryString(arr1[i]).replace(' ', '0')));
            String spot2 = String.format("%" + n + "s", String.valueOf(Integer.toBinaryString(arr2[i]).replace(' ', '0')));

            for (int j = 0; j < n; j++) {
                char n1 = spot1.charAt(j);
                char n2 = spot2.charAt(j);

                if (n1 == '1' || n2 == '1') {
                    sb.append('#');
                } else {
                    sb.append(' ');
                }
            }

            answer[i] = sb.toString();
        }

        return answer;
    }
}